pkg/validation/options.go

```diff
@@ -239,6 +239,8 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 	p.ValidateURL, msgs = parseURL(o.ValidateURL, "validate", msgs)
 	p.ProtectedResource, msgs = parseURL(o.ProtectedResource, "resource", msgs)
 
+	p.SetAllowedGroups(o.AllowedGroups)
```

Comment:
Unrelated to this PR.   Could you please handle the optional **groups** scope .   Many  providers does not supports thats "We get Invalid_scope error" 

`
			// if len(o.AllowedGroups) > 0 {
			// 	o.Scope += " groups"
			// }
`

Reply:
It only adds ` groups` to the default scope if you didn't set one yourself.  If your provider has issues with `groups` in the scope, just set the `--scope` configuration option yourself to what you need:

https://github.com/oauth2-proxy/oauth2-proxy/blob/1fe0cfc6f74db2543d6e8f608b0c8b0ddc18bca8/pkg/validation/options.go#L153

Reply:
Yeah !!! I did that .  I thought of getting your opinion as well.  Thanks

Reply:
IdPs are all over the place in how they implement `groups` since it isn't an official spec.

We erred toward adding the `groups` scope by default if a user set `allowed-groups` to support the major OIDC IdPs and stave off user issues where they asked for `allowed-groups` authorization logic, but didn't get any (because under the hood their IDTokens wouldn't have a `groups` claim if we didn't set the `groups` scope).